### DISCUSSION TITLE
Class components

### DISCUSSION THREAD

**Discussion by geenva** [vercel/next.js#23548] [Help] [answered]
I'm sorry if this is a frequently asked question, but is it possible/how can I use class components in Next?  I've searched it up, but found unrelated answers about something called `useRouter()`. Thanks!

**Comment by bendman** [comment, marked as answer]
@geenva You can see an example of a class component used for a Page, including fetching data, on [this page](https://nextjs.org/docs/api-reference/data-fetching/getInitialProps). If you need access to the router object you can use the [`withRouter`](https://nextjs.org/docs/api-reference/next/router#withrouter) function as a HOC, like this:

```jsx
import { withRouter } from 'next/router'

class MyPage extends React.Component {
  render() {
    return <div>You are at the path {this.props.router.pathname}</div>
  }
}

export default withRouter(MyPage)
```

**Reply by geenva** [reply]
Thank you!
